packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceAddUser.tsx

```diff
@@ -86,25 +94,33 @@ export const ShareSpaceAddUser: FC<ShareSpaceAddUserProps> = ({
         { keepPreviousData: true },
     );
 
-    const organizationUsers = useMemo(
-        () => infiniteOrganizationUsers?.pages.map((p) => p.data).flat(),
-        [infiniteOrganizationUsers?.pages],
-    );
+    useEffect(() => {
+        const allPages =
+            infiniteOrganizationUsers?.pages.map((p) => p.data).flat() ?? [];
+
+        setAllSearchedOrganizationUsers((previousState) =>
```

Comment:
I'm not fully understand what this is doing, but perhaps we should at least add acomment explaining what this is fixing.

Reply:
@rephus Code did not take into account how React Query caches data. Since the search query is part of the react query's cache key, every time the search query changes, the results returned by the hook exclude those from the previous search query.

This causes an issue where the autocomplete can display search results, but when you click on a result, the query resets to an empty string, loading a completely new set of results. As a result, the selected value disappears, as described in the original issue ticket.

This change aggregates all values returned by the hook into a state, keeping track of all users that have ever appeared in the search results, regardless of the search query.

I’ve included the updated comment.
hope it’s clearer now!